gson/src/main/java/com/google/gson/internal/bind/TypeAdapters.java

```diff
@@ -800,6 +802,14 @@ public EnumTypeAdapter(Class<T> classOfT) {
             for (String alternate : annotation.alternate()) {
               nameToConstant.put(alternate, constant);
             }
+            String regex = annotation.pattern();
+            if (regex != "") {
```

Comment:
Yes, thought about that. However, this is getting the annotation value, which is defaulted to "", so used ==. 
I will switch to equals()